server/protocols/webadmin/src/test/java/org/apache/james/webadmin/routes/DomainRoutesTest.java

```diff
@@ -0,0 +1,341 @@
+import static org.mockito.Mockito.when;
+
+import java.net.InetAddress;
+
+import org.apache.james.dnsservice.api.DNSService;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.domainlist.memory.MemoryDomainList;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.http.ContentType;
+
+import de.bechte.junit.runners.context.HierarchicalContextRunner;
+
+@RunWith(HierarchicalContextRunner.class)
+public class DomainRoutesTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DomainRoutesTest.class);
+    public static final String DOMAIN = "domain";
+
+    private WebAdminServer webAdminServer;
+
+    private void createServer(DomainList domainList) throws Exception {
+        webAdminServer = new WebAdminServer(new DomainRoutes(domainList, new JsonTransformer()));
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.port = webAdminServer.getPort();
+        RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+    }
+
+    @After
+    public void stop() {
+        webAdminServer.destroy();
+    }
+
+    public class NormalBehaviour {
+
+        @Before
+        public void setUp() throws Exception {
+            DNSService dnsService = mock(DNSService.class);
+            when(dnsService.getHostName(any())).thenReturn("localhost");
+            when(dnsService.getLocalHost()).thenReturn(InetAddress.getByName("localhost"));
+
+            MemoryDomainList domainList = new MemoryDomainList();
+            domainList.setDNSService(dnsService);
+            domainList.setLog(LOGGER);
+            domainList.setAutoDetectIP(false);
+            createServer(domainList);
+        }
+
+        @Test
+        public void getDomainsShouldBeEmptyByDefault() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .get(DomainRoutes.DOMAINS)
+            .then()
+                .statusCode(200)
+                .body(is("[]"));
+        }
+
+        @Test
+        public void postShouldBeOk() {
+            given()
+                .accept(ContentType.JSON)
+                .contentType(ContentType.JSON)
+            .when()
+                .post(DomainRoutes.DOMAIN + DOMAIN)
+            .then()
+                .statusCode(200);
+        }
+
+        @Test
+        public void getDomainsShouldDisplayAddedDomains() {
+            given()
+                .accept(ContentType.JSON)
```

Comment:
rest-assured allow configure default content type, that would make the code much compact